ttnn/cpp/ttnn/operations/moreh/moreh_fold/device/fold_device_operation.hpp

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/device_operation.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+#include "ttnn/tensor/types.hpp"
+
+namespace ttnn::operations::moreh::moreh_fold {
+
+struct MorehFoldOperation {
+    struct operation_attributes_t {
+        const std::vector<uint32_t> output_size;
+        const std::vector<uint32_t> kernel_size;
+        const std::vector<uint32_t> dilation;
+        const std::vector<uint32_t> padding;
+        const std::vector<uint32_t> stride;
+        const MemoryConfig output_memory_config;
```

Comment:
"memory_config" is convention
Should use that name throughout for this operation.

Reply:
Resolved in last commit.